Marlin/src/gcode/bedlevel/G35.cpp

```diff
@@ -91,8 +91,8 @@ void GcodeSuite::G35() {
   // Disable duplication mode on homing
   TERN_(HAS_DUPLICATION_MODE, set_duplication_enabled(false));
 
-  // Home all before this procedure
-  home_all_axes();
+  // Home only Z axis when X and Y is trusted, otherwise all axes, if needed before this procedure
+  if (!all_axes_trusted()) queue.inject_P(PSTR("G28Z"));
```

Comment:
Not sure if this is the right way to home Z, so I'm up for the corrections.

Reply:
From within a G-code handler you can always use `process_subcommands_now` and `process_subcommands_now_P`.